src/services/datasource.ts

```diff
@@ -128,11 +130,14 @@ export class WrappedLokiDatasource extends RuntimeDataSource<DataQuery> {
 
   private getData(
     request: SceneDataQueryRequest,
-    ds: DataSourceWithBackend<DataQuery>,
+    ds: DataSourceWithBackend<LokiQuery>,
     subscriber: Subscriber<DataQueryResponse>
   ) {
-    // query the datasource and return either observable or promise
-    const dsResponse = ds.query(request);
+    // @ts-expect-error
+    const shardingEnabled = config.featureToggles.exploreLogsShardSplitting;
+
+    // Query the datasource and return either observable or promise
+    const dsResponse = isLogsRequest(request) || !shardingEnabled ? ds.query(request) : runShardSplitQuery(ds, request);
```

Comment:
Exclude requests that contain `avg_over_time` expressions?

Reply:
Oh yes, definitely!

Reply:
Done.